scripts/upgrade

```diff
@@ -150,6 +150,17 @@ ynh_local_curl $installUrl "install=2" "db_type=mysqli" "hostname=localhost" "db
 # or on any installation that is accessible via the Internet.
 ynh_secure_remove "$final_path/admin"
 
+#=================================================
+# SETUP SSOWAT
+#=================================================
+ynh_script_progression --message="Configuring SSOwat..." --weight=1
+
+# Make app public if necessary or protect it
+if [ $is_public -eq 0 ]
```

Comment:
You shouldn't use `is_public` because permission can be changed by the admin of the YNH server. Please, before adding `visitors` to the permission, check if it's already in the permission with: https://github.com/YunoHost/yunohost/blob/1279755463100e6c4a0ba7d4b50f8f39658c8eac/data/helpers.d/setting#L308, and remove it if it was

Reply:
Will it make sense to reverse the permission granted to `visitors` on L136 with this? :
`ynh_permission_update --permission "main" --remove "visitors"`

Reply:
I think you should:

line 136:
```
ynh_permission_has_user --permission="main" --user="visitors"
contains_visitors=$?

if [ $contains_visitors -eq 1 ]
then
    ynh_permission_update --permission "main" --add "visitors"
fi
```

then here:
```
if [ $contains_visitors -eq 1 ]
then
    ynh_permission_update --permission "main" --remove "visitors"
fi
```

In fact, with the new permission system, we shouldn't store `is_public` anymore, and just look if `visitors` is present or not.